src/Controls/src/Core/Hosting/AppHostBuilderExtensions.cs

```diff
@@ -0,0 +1,257 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Maui.Controls.Handlers;
+using Microsoft.Maui.Controls.Handlers.Items;
+using Microsoft.Maui.Controls.Shapes;
+using Microsoft.Maui.Dispatching;
+using Microsoft.Maui.Handlers;
+using Microsoft.Maui.Hosting;
+using Microsoft.Maui.Platform;
+
+#if ANDROID
+using Microsoft.Maui.Controls.Handlers.Compatibility;
+using Microsoft.Maui.Controls.Compatibility.Platform.Android;
+#elif WINDOWS
+using ResourcesProvider = Microsoft.Maui.Controls.Compatibility.Platform.UWP.WindowsResourcesProvider;
+using Microsoft.Maui.Controls.Compatibility.Platform.UWP;
+#elif IOS || MACCATALYST
+using Microsoft.Maui.Controls.Compatibility.Platform.iOS;
+using Microsoft.Maui.Controls.Handlers.Compatibility;
+#elif TIZEN
+using Microsoft.Maui.Controls.Handlers.Compatibility;
+using Microsoft.Maui.Controls.Compatibility.Platform.Tizen;
+#endif
+
+namespace Microsoft.Maui.Controls.Hosting;
+
+public static partial class AppHostBuilderExtensions
```

Comment:
This class used to be in the Controls.Xaml assembly, however all of this was meant to be in Controls.Core. There is 2 lines in the old file that is referencing XAML types, so we rather move this logic into Controls.Core and call it from Controls.Xaml.